### PR TITLE
chore(jsdoc): toast jsdoc and improvements

### PR DESCRIPTION
**Pull request by anuraghazra**

### CODE REVIEW

File: src/utils/index.ts [hunk 1]
```diff
@@ -45,10 +45,9 @@ export function isTouch() {
   );
 }
 
-export const dataAttr = (condition: boolean | undefined) =>
+export const dataAttr = (condition?: boolean) =>
   (condition ? "" : undefined) as Booleanish;
 
-export const ariaAttr = (condition: boolean | undefined) =>
-  condition ? true : undefined;
+export const ariaAttr = (condition?: boolean) => (condition ? true : undefined);
```
Comment (navin-moorthy) [inline src/utils/index.ts:51]: Why this change 🤔 , any reasons?

I think line 51 can remain the same.

Comment (anuraghazra) [reply src/utils/index.ts:51]: Is this intentional that you used type `boolean | undefined`??

because the difference between `boolean | undefined` vs `condtion?: boolean` is first one expects to be undefined or boolean value and the argument cannot be empty.

but `condition?: boolean` argument can be empty and can pass undefined.

image
image

Comment (navin-moorthy) [reply src/utils/index.ts:51]: Based on the usage, both should be `boolean | undefined`.

We need the `condition` in both utils to be passed

Comment (navin-moorthy) [reply src/utils/index.ts:51]: I made a mistake, just now gave it a thought !!

Comment (anuraghazra) [reply src/utils/index.ts:51]: boolean | undefined is totally fine if you intentionally want to restrict the arguments but both would not effect any logic behaviour since if we don't provide the argument it will be undefined anyways.

Comment (navin-moorthy) [reply src/utils/index.ts:51]: Yeah, it would not affect the behaviour, we are just making it compulsory to pass a condition.

File: src/toast/ToastProvider.tsx [hunk 2]
```diff
@@ -3,6 +3,7 @@ import ReactDOM from "react-dom";
 import { canUseDOM } from "reakit-utils";
 import { createContext } from "@chakra-ui/utils";
```
Comment (navin-moorthy) [inline src/toast/ToastProvider.tsx:4]: Extract the utils from chakra - https://github.com/chakra-ui/chakra-ui/blob/develop/packages/utils/src/react-helpers.ts 🤔 to our local utils?

File: src/toast/ToastController.tsx [hunk 3]
```diff
@@ -2,10 +2,29 @@ import React from "react";
 import { useTimeout } from "@chakra-ui/hooks";
```
Comment (navin-moorthy) [inline src/toast/ToastController.tsx:2]: Can we use their hooks package OR should we also localize this ?

Comment (navin-moorthy) [reply src/toast/ToastController.tsx:2]: Can we reexport form chakra/hook from our newly created "hooks" folder 🤔

Comment (anuraghazra) [reply src/toast/ToastController.tsx:2]: Depends on if we are using chakra hooks anywhere else, if this is the only time we are using the package we can extract it out

Comment (navin-moorthy) [reply src/toast/ToastController.tsx:2]: We are using many hooks from chakra.

Comment (anuraghazra) [reply src/toast/ToastController.tsx:2]: Then i think we can keep it, since it would be well tested & updated.

Comment (navin-moorthy) [reply src/toast/ToastController.tsx:2]: Okay, okay !!

File: src/toast/ToastProvider.tsx (hunk 2 above)
Comment (navin-moorthy) [reply src/toast/ToastProvider.tsx:4]: What about this @anuraghazra  ?

File: src/toast/ToastController.tsx (hunk 3 above)
Comment (anuraghazra) [reply src/toast/ToastController.tsx:2]: Yeah

File: src/toast/ToastProvider.tsx (hunk 2 above)
Comment (anuraghazra) [reply src/toast/ToastProvider.tsx:4]: I will extract it out...

**Review by navin-moorthy — APPROVED** [review]
LGTM 🎉
